programs/install/Install.cpp

```diff
@@ -446,8 +446,8 @@ int mainEntryClickHouseInstall(int argc, char ** argv)
                 fs::path ulimits_file = ulimits_dir / fmt::format("{}.conf", user);
                 fmt::print("Will set ulimits for {} user in {}.\n", user, ulimits_file.string());
                 std::string ulimits_content = fmt::format(
-                    "{0}\tsoft\tnofile\t262144\n"
-                    "{0}\thard\tnofile\t262144\n", user);
+                    "{0}\tsoft\tnofile\t1048576\n"
+                    "{0}\thard\tnofile\t1048576\n", user);
```

Comment:
I think nowadays systemd is kind of defacto standard.

And for systemd it is better to use `LimitNOFILE=1048576` systemd unit directive.
Yes systemd does interpret `/etc/security` too, but only if it has been configured with `PAM` (although I guess all major distributives does so), and limits from the unit (`LimitNOFILE`, ...) will override these.

And right now limit in systemd is 500K - https://github.com/ClickHouse/ClickHouse/blob/79bb44ad80ca916f42d260ccc0e10f4675203a88/packages/clickhouse-server.service#L22